fix(react): enable event names with forward slash

alexprudhomme opened the pull request:
fix #473 

## Pull request checklist

<!-- Please note that this repository is largely maintained for the purposes of supporting the Ionic Framework -->
<!-- As a result, we may not immediately (or ever) get around to reviewing pull requests that do not support the Framework -->

Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Build (`npm run build`) was run locally for affected output targets
- [x] Tests (`npm test`) were run locally and passed
- [x] Prettier (`npm run prettier`) was run locally and passed

## Pull request type

<!-- Please do not submit updates to dependencies unless it fixes an issue. -->

<!-- Please try to limit your pull request to one type, submit multiple pull requests if needed. -->

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

<!-- Please describe the current behavior that you are modifying. -->

<!-- Issues are required for both bug fixes and features. -->

Issue URL:

## What is the new behavior?

<!-- Please describe the behavior or changes that are being added by this PR. -->

- enables event names with forward slash in react-output-target
-
-

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

<!-- If this introduces a breaking change, please describe the impact and migration path for existing applications below. -->

## Other information

<!-- Any other information that is important to this PR such as screenshots of how the component looks before and after the change. -->

tanner-reits reviewed (changes requested):
@alexprudhomme I tested this out with your `/` event naming, but doesn't seem to work with that convention. As far as I know, React requires strict camel-casing for event handlers, so if you wanted to have event names with a `/` in them, you would need to update the logic [here](https://github.com/ionic-team/stencil-ds-output-targets/blob/main/packages/react-output-target/src/react-output-target/utils/string-utils.ts#L10) to also perform those replacements

alexprudhomme commented:
> @alexprudhomme I tested this out with your `/` event naming, but doesn't seem to work with that convention. As far as I know, React requires strict camel-casing for event handlers, so if you wanted to have event names with a `/` in them, you would need to update the logic [here](https://github.com/ionic-team/stencil-ds-output-targets/blob/main/packages/react-output-target/src/react-output-target/utils/string-utils.ts#L10) to also perform those replacements

Hey, thanks for the useful tip. I changed the implementation and added a test. Let me know if you need anything more.

tanner-reits reviewed (changes requested):
@alexprudhomme Functionality looks good! Only thing I would like to see changed is to leave the `kebabToCamelCase` function as it was (in case we have need for that utility later), and move the `/` replacement to a separate utility function that can be called in `eventListenerName`.

tanner-reits commented on packages/react-output-target/src/react-output-target/utils/string-utils.ts:
```diff
@@ -7,4 +7,9 @@ export const kebabToPascalCase = (str: string) =>
 export const kebabToCamelCase = (str: string) => str.replace(/-([a-z])/g, (_, letter) => letter.toUpperCase());
 
-export const eventListenerName = (eventName: string) => kebabToCamelCase(`on-${eventName}`);
+const dashesToCamelCase = (str: string) => str.replace(/\/([a-z])/g, (_, letter) => letter.toUpperCase());
```
```suggestion
const slashesToCamelCase = (str: string) => str.replace(/\/([a-z])/g, (_, letter) => letter.toUpperCase());
```

tanner-reits commented on packages/react-output-target/src/react-output-target/utils/string-utils.ts:
```diff
@@ -7,4 +7,9 @@ export const kebabToPascalCase = (str: string) =>
+const dashesToCamelCase = (str: string) => str.replace(/\/([a-z])/g, (_, letter) => letter.toUpperCase());
+
+export const eventListenerName = (eventName: string) => {
+  const dashedConverted = dashesToCamelCase(eventName);
+  return kebabToCamelCase(`on-${dashedConverted}`);
```
```suggestion
  const slashesConverted = slashesToCamelCase(eventName);
  return kebabToCamelCase(`on-${slashesConverted}`);
```

tanner-reits reviewed (changes requested):
Just some naming fixes, then g2g

alexprudhomme commented:
@tanner-reits 
Hi, 
Will you be able to make a release soon with this fix ? Last one was on august 28th. 
Thanks !

christian-bromann commented:
@alexprudhomme I am working on it as we speak.
